### ISSUE TITLE
CORS replication problem on 6.0

### ISSUE THREAD

**Issue by jefbarn** [pouchdb/pouchdb#5644]
### Issue
After updating to PouchDB 6.0.4 seeing a CORS related failure when replicating a CouchDB 2.0rc3 host to browser (tested on both Chrome 52 and Firefox 48)
Verified that going back to 5.4.5 cleared the issue.

Some log excerpts:
 OPTIONS https://my.url.com/couchdb/my_database/_local%2F2r6IyXA0WQtv2tXR8NA88A%3D%3D 
2016-09-06 12:57:09.451 102:1 XMLHttpRequest cannot load 

```
replication error Object {status: 0, name: "unknown", message: "writeCheckpoint completed with error", result: Object}message: "writeCheckpoint completed with error"name: "unknown"result: Objectstatus: 0__proto__: Object(anonymous function) @ app.be400f.loader.js:119111EventEmitter.emit @ app.be400f.loader.js:8172backOff @ app.be400f.loader.js:113866completeReplication @ app.be400f.loader.js:114141abortReplication @ app.be400f.loader.js:114114onCheckpointError @ app.be400f.loader.js:114302(anonymous function) @ app.be400f.loader.js:114007ZoneDelegate.invoke @ app.be400f.loader.js:80417Zone.run @ app.be400f.loader.js:80310(anonymous function) @ app.be400f.loader.js:80676ZoneDelegate.invokeTask @ app.be400f.loader.js:80450Zone.runTask @ 
```

#5455 related?

**Comment by willholley** [comment]
@jefbarn can you find the corresponding error in your CouchDB log? I wonder if there is a double URL-encoding issue.

**Comment by jefbarn** [comment]
I see the checkpoint GET, but it doesn't look like couchDB is logging the OPTIONS requests.
```
[notice] 2016-09-07T17:32:26.291000Z couchdb@localhost <0.14135.16> e382114447 ns-db-agi-dev01.gci.com:5984 10.180.185.84:3519 undefined GET /daylight/_local/yEdoIQyc6xofQ_igEfLp1A==? 404 ok 3
```

I'll try to see if there is a setting to enable this in the CouchDB config.

**Comment by jefbarn** [comment]
Looks like this is something with my reverse proxy.  I tested connecting direct to CouchDB and it's working fine.

**Comment by jefbarn** [comment]
I think we need to be using encodeDocId instead of encodeURIComponent which doesn't url encode the '/'.  I remember doing this gives most reverse proxies a conniption.

**Comment by jefbarn** [comment]
Tested using encodeDocId works both with and without reverse proxy on Couch 2.0

**Comment by nolanlawson** [comment]
fixed in #5652
